v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/SpannerChangeStreamsToBigQueryIT.java

```diff
@@ -0,0 +1,224 @@
+        BigQueryResourceManager.builder(testName, PROJECT, credentials).build();
+  }
+
+  @After
+  public void tearDown() {
+    ResourceManagerUtils.cleanResources(spannerResourceManager, bigQueryResourceManager);
+  }
+
+  @Test
+  public void testSpannerChangeStreamsToBigQueryBasic() throws IOException {
+    String spannerTable = testName + RandomStringUtils.randomAlphanumeric(1, 5);
+    String createTableStatement =
+        String.format(
+            "CREATE TABLE %s (\n"
+                + "  Id INT64 NOT NULL,\n"
+                + "  FirstName String(1024),\n"
+                + "  LastName String(1024),\n"
+                + ") PRIMARY KEY(Id)",
+            spannerTable);
+
+    String cdcTable = spannerTable + "_changelog";
+    spannerResourceManager.executeDdlStatement(createTableStatement);
+    String createChangeStreamStatement =
+        String.format(
+            "CREATE CHANGE STREAM %s_stream FOR %s OPTIONS (value_capture_type = 'NEW_ROW')",
+            testName, spannerTable);
+    spannerResourceManager.executeDdlStatement(createChangeStreamStatement);
+    bigQueryResourceManager.createDataset(REGION);
+
+    Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder = Function.identity();
+    ;
+
+    launchInfo =
+        launchTemplate(
+            paramsAdder.apply(
+                LaunchConfig.builder(testName, specPath)
+                    .addParameter("spannerProjectId", PROJECT)
+                    .addParameter("spannerInstanceId", spannerResourceManager.getInstanceId())
+                    .addParameter("spannerDatabase", spannerResourceManager.getDatabaseId())
+                    .addParameter(
+                        "spannerMetadataInstanceId", spannerResourceManager.getInstanceId())
+                    .addParameter("spannerMetadataDatabase", spannerResourceManager.getDatabaseId())
+                    .addParameter("spannerChangeStreamName", testName + "_stream")
+                    .addParameter("bigQueryDataset", bigQueryResourceManager.getDatasetId())
+                    .addParameter("rpcPriority", "HIGH")
+                    .addParameter("disableDlqRetries", "true")));
+
+    assertThatPipeline(launchInfo).isRunning();
+
+    int key = nextValue();
+    String firstName = UUID.randomUUID().toString();
+    String lastName = UUID.randomUUID().toString();
+    List<Pair> colValPairs =
+        Arrays.asList(new Pair("FirstName", firstName), new Pair("LastName", lastName));
+    Mutation expectedData = generateTableRow(spannerTable, key, colValPairs);
+    spannerResourceManager.write(Collections.singletonList(expectedData));
+    String query = queryCdcTable(cdcTable, key);
+    waitForQueryToReturnRows(query, 1, true);
+
+    TableResult tableResult = bigQueryResourceManager.runQuery(query);
+    assertEquals(1, tableResult.getTotalRows());
+    for (FieldValueList row : tableResult.iterateAll()) {
+      assertTrue(validateChangeLogTableRow(row, colValPairs));
+    }
```

Comment:
Same here
```suggestion
    TableResult tableResult = bigQueryResourceManager.readTable(cdcTable);
    List<Map<String, Object>> expectedRecords = 
        SpannerAsserts.mutationsToRecords(List.of(expectedData));
    assertThatBigQueryRecords(tableResult).hasRecordsUnordered(expectedRecords);
```

Reply:
Thanks for sharing this test util function! I tried using this but ended up with above failure
![image](https://github.com/GoogleCloudPlatform/DataflowTemplates/assets/43364656/2d394714-d208-4aaf-a2bb-d6284e822f2e)
As you can see we have a bunch of metadata columns in cddTable, and we only care about the values for the actual mapping columns with the source spanner table. In addition, the metadata columns `_metadata_spanner_commit_timestamp` and `_metadata_bigquery_commit_timestamp` are in-place generated by spanner and bigquery. We can't really get the exact expect values for them ahead of time. Thus I thought it would be easier to just compare the column values of the actual spanner table columns. Let me know if you have better suggestions on handling this, thank!

Reply:
Could also do
```
TableResult tableResult = bigQueryResourceManager.readTable(cdcTable);
    List<Map<String, Object>> expectedRecords = 
        SpannerAsserts.mutationsToRecords(List.of(expectedData))
            .forEach(record -> assertThatBigQueryRecords(tableResult).hasRecordSubset(record);
```
Which does a partial check to make sure each BQ row contains _at least_ the columns/values from the expected spanner row

Reply:
I tried `.hasRecordSubset` as you suggested but still encounters the same failure:
![image](https://github.com/GoogleCloudPlatform/DataflowTemplates/assets/43364656/a2659399-ac4a-44e9-b9c1-f52b8521041d)
Seems like `mutationsToRecords` does not guaranteed the ordinal position for the corresponding column. I built the spanner mutation in the order of id, first name and last name. However the mutationsToRecords returns it in the order of first name, id and last name, which is in alphabetical order. `hasRecordSubset` seems can't handle unordered situation like `hasRecordsUnordered `. Let me know if you are ok with keeping current column validation logic or I can also manually re-sort `List<Map<String, Object>>` in original column ordinal position order in spanner table. Thanks!

Reply:
Hm that's unfortunate. I Should probably update that. But for this PR, that's totally fine